File: DSAjavascript/BinarySearch/ArrangingCoin.js
/**
 * problem no:- 441 
 * leetcode solution 
 * link Here(https://leetcode.com/problems/arranging-coins/submissions/1228015361/)
 */

/**
 * @param {number} n
 * @return {number}
 */
var arrangeCoins = function(n) {

    let lo = 1; 
    let hi = n;
    let ans =1;

    while(lo<=hi){
        let mid = lo +Math.floor((hi-lo)/2);
         let value = Math.floor(((mid)*(mid+1))/2);

         //if(value == n)
         if(value <= n){
            ans = mid;
            lo=mid+1;
         }else{

            hi = mid -1;
         }
    }

        return ans;

    
    
};

File: DSAinJava/search.java/ArrangingCoin.java
/**
 * problem no :- 441 leetrcode 
 * 
 * leetcode solution link(https://leetcode.com/problems/arranging-coins/submissions/1228015361/)
 * 
 */


public class ArrangingCoin {
    
        public int arrangeCoins(int n) {
    
            
        long  lo = 1; 
        long   hi = n;
        long  ans =1;
    
        while(lo<=hi){
            long  mid = lo+ (hi-lo)/2;
             long  value =(mid*(mid+1))/2 ;
    
             //if(value == n)
             if(value <= n){
                ans = mid;
                lo=mid+1;
             }else{
    
                hi = mid -1;
             }
        }
    
            return (int)ans;
    
            
        }
    
}
